subxt/src/client/online_client.rs

```diff
@@ -90,22 +123,69 @@ impl<T: Config> OnlineClient<T> {
     pub async fn from_rpc_client<R: RpcClientT>(
         rpc_client: Arc<R>,
     ) -> Result<OnlineClient<T>, Error> {
-        let rpc = Rpc::new(rpc_client);
+        OnlineClient::from_rpc_client_at(rpc_client, None).await
+    }
 
+    /// Construct a new [`OnlineClient`] by providing an underlying [`RpcClientT`]
+    /// implementation to drive the connection, as well as a block hash from which to
+    /// obtain the metadata information from.
+    ///
+    /// # Warning
+    ///
+    /// - If you use this function, subxt will be unable to work with blocks that aren't
+    ///   compatible with the metadata at the provided block hash, and if the block hash is
+    ///   not recent, things like subscribing to the head of the chain and submitting
+    ///   transactions will likely fail or encounter errors.
+    /// - Subxt does not support blocks using pre-V14 metadata and will error if you attempt
+    ///   to instantiate a client at such a block.
+    pub async fn from_rpc_client_at<R: RpcClientT>(
+        rpc_client: Arc<R>,
+        block_hash: Option<T::Hash>,
+    ) -> Result<OnlineClient<T>, Error> {
+        let rpc = Rpc::<T>::new(rpc_client.clone());
         let (genesis_hash, runtime_version, metadata) = future::join3(
             rpc.genesis_hash(),
-            rpc.runtime_version(None),
-            rpc.metadata(None),
+            // This _could_ always be the latest runtime version, but offhand it makes sense to keep
```

Comment:
not sure what you mean here? Are you saying that it could be the latest because it doesn't impact anything downstream? like event decoding?

Reply:
I personally interpret the comment as the block hash (if provided) could already contain the latest metadata

Reply:
Hmm I would almost argue the comment makes things more confusing than no comment in that case

Reply:
yeah, I think you can remove this comment.

The docs and the function name itself makes this code obvious IMO

Reply:
Yeah I can remove it; what I meant was just that even if the metadata was at some specific block and the runtime version was always the latest, I don't think it would break anything related to working with older blocks (and would increase the chance that submitting transactions would work). But in the end I prefer to keep the two aligned always.